JournalCoreData/Model Controllers/EntryController.swift

```diff
@@ -109,7 +109,7 @@ class EntryController {
                 return
             }
 
-            let moc = CoreDataStack.shared.mainContext
+            let moc = CoreDataStack.shared.container.newBackgroundContext()
```

Comment:
Consider moving your completion handlers in the perform block to the main queue to avoid race calling anything on a background queue